scripts/core/editor3/components/DraftSpellchecker.jsx

```diff
@@ -0,0 +1,244 @@
+/**
+ * @ngdoc React
+ * @module superdesk.core.editor3
+ * @name DraftSpellchecker
+ * @param {String} language the current spellchecker language
+ * @description check the text and shos the inline spellchecker context menu
+ */
+
+import React from 'react';
+import {InlineSpellcheckMenu} from '.';
+
+/** Custom overrides for "error" style. */
+const styleMap = {
+    ERROR: {
+        borderBottomWidth: '1px',
+        borderBottomStyle: 'dotted',
+        borderBottomColor: 'red'
+    }
+};
+
+export class DraftSpellchecker {
+    constructor(language) {
+        this.language = language;
+        this.suggestions = {};
+        this.words;
+        this.ignoredWords;
+
+        this.setContextMenu;
+        this.closeContextMenu;
+        this.initDecorators;
+        this.onReplaceWord;
+
+        this.onIgnoreWord;
+        this.onAddWord;
+        this.getSuggestions;
+        this.getWords;
+        this.getIgnoredWords;
+
+        this.registerEditorCallbacks = this.registerEditorCallbacks.bind(this);
+        this.registerDictionaryCallbacks = this.registerDictionaryCallbacks.bind(this);
+
+        this.getContextMenu = this.getContextMenu.bind(this);
+        this.showContextMenu = this.showContextMenu.bind(this);
+
+        this.getDecorator = this.getDecorator.bind(this);
+        this.isCorrectWord = this.isCorrectWord.bind(this);
+
+        this.ignoreWord = this.ignoreWord.bind(this);
+        this.addWord = this.addWord.bind(this);
+        this.replaceWord = this.replaceWord.bind(this);
+    }
+
+    /**
+      * @ngdoc method
+      * @name DraftSpellchecker#registerEditorCallbacks
+      * @param {Function} setContextMenu context menu initialization callback
+      * @param {Function} closeContextMenu close context menu callback
+      * @param {Function} initDecorator set the decorator callback in order to force a full refresh
+      * @param {Function} onReplaceWord replace word callback
+      * @description Register editor related actions
+      */
+    registerEditorCallbacks(setContextMenu, closeContextMenu, initDecorators, onReplaceWord) {
```

Comment:
I would remove this pattern. In my opinion it indicates that separation of concerns could be better. All of these functions that are _registered_ here could (and should) live in the context menu.

The context menu can be given access to `editorState` and it can use it to obtain anything it needs.

Additionally, the context menu (as it already is a React component) can be added into the `render()` method of editor, just like `Toolbar` has been.